plugins/modem-manager/fu-mm-device.c

```diff
@@ -1755,31 +1756,94 @@ fu_mm_device_setup_branch_at(FuMmDevice *self, GError **error)
 	if (locker == NULL)
 		return FALSE;
 
-	/* firmware branch AT command may fail if not implemented,
-	 * clear error if not supported */
-	if (self->branch_at != NULL) {
-		g_autoptr(GError) error_branch = NULL;
-		if (!fu_mm_device_at_cmd(self, self->branch_at, TRUE, &error_branch))
-			g_debug("unable to get firmware branch: %s", error_branch->message);
-	}
+	if (!fu_mm_device_at_cmd(self, self->branch_at, TRUE, error))
+		return FALSE;
+
+	if (fu_device_get_branch(self) != NULL)
+		g_debug("using firmware branch: %s", fu_device_get_branch(self));
+	else
+		g_debug("using firmware branch: default");
 
-	/* success */
 	return TRUE;
 }
 
+static void
+fu_mm_device_setup_secboot_status_quectel(FuMmDevice *self)
+{
+	if (self->port_at != NULL) {
+		g_autoptr(FuDeviceLocker) locker = NULL;
+		g_autoptr(GError) error_local = NULL;
+
+		/* Create IO channel to send AT commands to the modem */
+		locker = fu_device_locker_new_full(self,
+						   (FuDeviceLockerFunc)fu_mm_device_io_open,
+						   (FuDeviceLockerFunc)fu_mm_device_io_close,
+						   &error_local);
+		if (locker == NULL) {
+			g_debug("failed to open AT port: %s", error_local->message);
+			return;
+		}
+
+		if (!fu_mm_device_at_cmd(self, "AT+QSECBOOT=\"status\"", TRUE, &error_local))
+			g_debug("failed to send AT+QSECBOOT command: %s", error_local->message);
+	} else {
+		const gchar *version = fu_device_get_version(FU_DEVICE(self));
+		g_autofree gchar *name = NULL;
```

Comment:
I think if AT+QSECBOOT fails, we want to fall back to the table; i.e. return TRUE if fu_mm_device_at_cmd succeeded, otherwise fall back to the lower parts. I've added you to the committers, so if you push your branch to the origin server I can fix it up for you.

Reply:
thanks! just pushed the latest edit to the origin